frame/system/src/mocking.rs

```diff
@@ -29,6 +29,19 @@ pub type MockUncheckedExtrinsic<T, Signature = (), Extra = ()> = generic::Unchec
 
 /// An implementation of `sp_runtime::traits::Block` to be used in tests.
 pub type MockBlock<T> = generic::Block<
-	generic::Header<<T as crate::Config>::BlockNumber, sp_runtime::traits::BlakeTwo256>,
+	generic::Header<u64, sp_runtime::traits::BlakeTwo256>,
+	MockUncheckedExtrinsic<T>,
+>;
+
+/// An implementation of `sp_runtime::traits::Block` to be used in tests with u32 BlockNumber type.
+pub type MockBlockU32<T> = generic::Block<
```

Comment:
Are these really needed?

Reply:
If it is a handful of tests that unreasonable set the block number type to u32 or u128, I would just change those to u64. 

Although, in production, the block number type is u32, so perhaps we should refactor all of them to just be u32, increasing the overlap between testing and production.

Reply:
There were mostly needed for Cumulus and Polkadot which also used different block number types. Moving them to u32 was adding a number of changes which would have been difficult to review as a part of this PR. I can probably take that as a follow-up.